Add ATR Butterfly-Valve as example (#579)
idaholab/virtual_test_bed: doc/content/research_reactors/atr/butterfly_valve/butterfly_valve_model.md

# ATR Butterfly-Valve Model using MOOSE Navier-Stokes module

*Contact: Daniel Yankura, daniel.yankura@inl.gov

*Model link: [ATR Butterfly-Valve Model](https://github.com/idaholab/virtual_test_bed/tree/main/research_reactors/atr/butterfly_valve)*

!tag name=ATR Butterfly-Valve Model

>      description=3D Steady state CFD model of a butterfly-vale used at ATR, using the Navier-Stokes module

Author: If I'm including all the meshes I used (coarse, medium, and fine), should I still specify coarse in the description?

>      description=3D Steady state CFD model of a butterfly-vale used at ATR, using the Navier-Stokes module

Reviewer: I think so. How many cells did the starCCM model use?

>      description=3D Steady state CFD model of a butterfly-vale used at ATR, using the Navier-Stokes module

Author: Not sure of the exact number but I believe it was in the 8-10 million range so much finer than anything I used.

>      description=3D Steady state CFD model of a butterfly-vale used at ATR, using the Navier-Stokes module

Reviewer: we can reach that with the new solver but not with the one you are using here. 

The text needs to reflect that, by mentioning coarse mesh thermal hydraulics and intermediate fidelity. 
The value of this approach is that it's usually enough to match experimental results, which you should show here

flow-rate, three meshes of different refinement levels were used. A closeup of the coarse 0$^\circ$ mesh is show below.

Due to convergence constraints a direct solve method was used, greatly increasing the computational resources needed.
Access to HPC resources will likely be necessary to run these simulations. It is also necessary to increase MOOSE's
default AD container size. More information on MOOSE's automatic differentiation system, and information on increases AD container size
when building can be found [here](https://mooseframework.inl.gov/automatic_differentiation/index.html).

!media atr/0deg_coarse_zoom_mesh.png
       style=width:50%;margin-left:auto;margin-right:auto
       id=0deg_bf_valve
       caption=A coarse mesh for a butterfly-valve in use at ATR. In this particular mesh the valve is fully closed.

!media atr/42deg_coarse_zoom_mesh.png
       style=width:50%;margin-left:auto;margin-right:auto
       id=42deg_bf_valve
       caption=A coarse mesh for a butterfly-valve in use at ATR. In this particular mesh the valve is opened to $42.7^\circ$.

# Butterfly-Valve Input

## Parameters

> The viscosity and density were set to match experimental values for water at the pressure levels measured in earlier experiments.

Author: The experimental pressure values are from an internal document but I can add a reference to it so INL people can check it out? I can definitely include a reference though to the viscosity table that I calculated it from.

to the desired level. 

!listing /research_reactors/atr/butterfly_valve/input_file/bf_valve_mixing_length_test.i block=Executioner language=cpp

### `Viscosity Rampdown`

The viscosity rampdown is controled but two input blocks, the `Functions` and `Materials`. The functions block determines how quickly the rampdown occurs while the materials block is what defines the
value of `mu` as show in the listing below. For these simlations, `mu` was first ramped down to a value of `1e-3 PaS`. Once it converged to that point, it was restarted and a new rampdown
function was used to quickly bring the viscosity down to `5.4e-4 PaS`.

!listing /research_reactors/atr/butterfly_valve/input_file/bf_valve_mixing_length_test.i block=Functions language=cpp

!listing /research_reactors/atr/butterfly_valve/input_file/bf_valve_mixing_length_test.i block=Materials language=cpp

# Results

Shown below is a velocity streamline plot for a select inlet velocity. Aside from general
trends such as high velocities at the top an bottom of the valve, important details like vortices forming downstream of the valve

> are able to be simulated despite the relative coarseness of the meshes.

Author: I ran the fine meshes I made in STAR and made a plot of experimental vs simulated including those STAR data points. I could include pictures from those runs?

I know there was a 3rd party engineering firm that did some other STAR simulations with very fine overset meshes where the valve actually moved. Since INL researchers didn't do those simulations I'm not sure what the process would be to add that to this example?

# ATR Butterfly-Valve Model using coarse mesh thermal hydraulics

*Contact: Daniel Yankura, daniel.yankura@inl.gov

*Model link: [ATR Butterfly-Valve Model](https://github.com/idaholab/virtual_test_bed/tree/main/research_reactors/atr/butterfly_valve)*

!tag name=ATR Butterfly-Valve Model
     description=3D Steady state coarse mesh CFD model of a butterfly-vale used at ATR, using the Navier-Stokes module
     pairs=reactor_type:LWR
                       geometry:valve
                       simulation_type:CFD
                       codes_used:MOOSE_NavierStokes
                       open_source:full
                       computing_needs:HPC
                       fiscal_year:2024
                       institution:INL

> The advanced test reactor (ATR) at INL employs a buttefly-valve to control pressure drop across the reactor core.

Reviewer: ```suggestion
The advanced test reactor (ATR) at INL employs a butterfly-valve to control pressure drop across the reactor core.
```

# Results

Shown below are velocity and pressure profiles for each angle and for select inlet velocities. These profiles were all taken from the finer mesh simulations. They display expected features
such as pressure and velocity peaks in the correct regions.   

!media atr/bf_valve_vel_and_pressure.png
       style=width:50%;margin-left:auto;margin-right:auto
       id=0deg_bf_valve_vel_and_pressure
       caption=Velocity and pressure profiles for each valve configuration.

Simulations were also performed on the fine mesh using STAR-CCM+ as a point of comparison. The pressure drop, resistance coefficient, and flow coefficient for all configurations (and for STAR-CCM+ simulations) is shown in the figure below. The experimental data was taken from an interal study by [!cite](ECAR52). Also included is a table with the pressure drop error and convergence information for the different levels of mesh refinement.
For most of the angles tested, simulations approached experimental value as finer meshes were used, with the only exception being the 22.5$^\circ$ configuration. However for that particular configuration simulation error relative to

> experamental values was already significantly smaller than other configurations.

Reviewer: ```suggestion
experimental values was already significantly smaller than other configurations.
```

# ATR Butterfly-Valve Model using MOOSE Navier-Stokes module

*Contact: Daniel Yankura, daniel.yankura@inl.gov

> *Model link: [ATR Butterfly-Valve Model](https://github.com/idaholab/virtual_test_bed/tree/devel/research_reactors/atr/butterfly_valve)*

Reviewer: ```suggestion
*Model link: [ATR Butterfly-Valve Model](https://github.com/idaholab/virtual_test_bed/tree/main/research_reactors/atr/butterfly_valve)*
```

!tag name=ATR Butterfly-Valve Model
     description=3D Steady state CFD model of a butterfly-vale used at ATR, using the Navier-Stokes module
     pairs=reactor_type:LWR
                       geometry:valve

>                        simulation_type:fluid_flow

Reviewer: let's try to fit in one of the existing categories. it s probably called cfd

                       codes_used:MOOSE_NavierStokes
                       open_source:full
                       computing_needs:HPC
                       fiscal_year:2024
                       institution:INL 

The advanced test reactor (ATR) at INL employs a buttefly-valve to control pressure drop across the reactor core. 
This simulation utilized the Navier-Stokes module in MOOSE for all simulations. 
Running this model will likely require HPC resources. The simulations setup will be described below.

# Physics Models and Boundary Conditions Used

> All simulations used the finite volume incompressible Navier-Stokes formulas. The pipe inlet was set as a

Reviewer: ```suggestion
All simulations used the finite volume discretization of the incompressible Navier-Stokes equations. The pipe inlet was set as a
```

constant inlet velocity that corresponded to the mass-flow rate from experimental data. The outlet pressure 
was arbitrarily set to 0. All walls (including those belonging to the valve) were set as no-slip boundaries.

# Simulation Setup

The valve was simulated for 5 different angles (between 0 and 42.7 degrees) where fully open is 90 degrees.

> Due to the highly turbulent nature of the simulations, the mixing-length turbulence model with an lu
> preconditioner was used. For each valve configuration and flow-rate, three meshes of different refinement levels

Reviewer: ```suggestion
Due to the highly turbulent nature of the simulations, the mixing-length turbulence model was used. For each valve configuration and flow-rate, three meshes of different refinement levels
```

Reviewer: LU is a numerical implementation detail, while mixing length is a physical model

were used. A closeup of the coarse 0 degree mesh is show below.

Due to convergence constraints a direct solve method was used, greatly increasing the computational resources needed.
Access to HPC resources will likely be necessary to run these simulations. It is also necessary to increase MOOSE's

> default AD container size. 

Reviewer: add a link

Author: Just checking, are you looking for a link to the hpc resources used, a link that explains how to increase AD container size, a link that explains more about direct solve methods, or something else?

Reviewer: to info on chaging the default AD container size

the other things could use links too
HPC resource could be a table instead

!media atr/0deg_coarse_zoom_mesh.png
       style=width:50%;margin-left:auto;margin-right:auto
       id=0deg_bf_valve
       caption=A coarse mesh for a butterfly-valve in use at ATR. In this particular mesh the valve is fully closed.

# Butterfly-Valve Input

## Parameters

```language=bash
  
 mu = 0.001 # viscosity [Pa*s]
 rho = 986.737 # density [kg/m^3]
 von_karman_const = 0.41
```

# Mesh

 
Because of the complex geometries involved, Cubit was used to generate meshes for all angles tested.

### `Variables`

test

!listing /research_reactors/atr/butterfly_valve/input_file/mixing_length_0deg_20k_gpm.i block=Variables language=cpp

### `AuxVariables`

!listing /research_reactors/atr/butterfly_valve/input_file/mixing_length_0deg_20k_gpm.i block=AuxVariables language=cpp

### `Aux Kernels`

!listing /research_reactors/atr/butterfly_valve/input_file/mixing_length_0deg_20k_gpm.i block=AuxKernels language=cpp

# Executioner

!listing /research_reactors/atr/butterfly_valve/input_file/mixing_length_0deg_20k_gpm.i block=Executioner language=cpp

> 

Reviewer: if you have a mesh refinement study it would be good to include it

Author: I added a reference to the Mesh section. It describes how to use mesh refinement to calculate self-convergence.

# Results

Shown below is a velocity streamline plot for a select inlet velocity for a select inlet velocity. Aside from general

> trends such as high velocities at the top an bottom of the valve, important details such as vorticies forming downstream of the valve

Reviewer: ```suggestion
trends such as high velocities at the top an bottom of the valve, important details such as vortices forming downstream of the valve
```
